captum/attr/_core/lime.py

```diff
@@ -0,0 +1,1148 @@
+            )
+            return interp_model
+
+    def _evaluate_batch(
+        self,
+        curr_model_inputs: List[TensorOrTupleOfTensorsGeneric],
+        expanded_target: TargetType,
+        expanded_additional_args: Any,
+        device: torch.device,
+    ):
+        model_out = _run_forward(
+            self.forward_func,
+            _reduce_list(curr_model_inputs),
+            expanded_target,
+            expanded_additional_args,
+        )
+        if isinstance(model_out, Tensor):
+            assert model_out.numel() == len(curr_model_inputs), (
+                "Number of outputs is not appropriate, must return "
+                "one output per perturbed input"
+            )
+        if isinstance(model_out, Tensor):
+            return model_out.flatten()
+        return torch.tensor([model_out], device=device)
+
+    def has_convergence_delta(self) -> bool:
+        return False
+
+    @property
+    def multiplies_by_inputs(self):
+        return False
+
+
+# Default transformations and methods
+# for Lime child implementation.
+
+
+def lasso_interpretable_model_trainer(
+    interpretable_inputs: Tensor, expected_outputs: Tensor, weights: Tensor, **kwargs
+):
+    try:
+        import sklearn
+        from sklearn import linear_model
+
+        assert (
+            sklearn.__version__ >= "0.23.0"
+        ), "Must have sklearn version 0.23.0 or higher to use "
+        "sample_weight in Lasso regression."
+    except ImportError:
+        raise AssertionError(
+            "Requires sklearn for default interpretable model training with"
+            " Lasso regression. Please install sklearn or use a custom interpretable"
+            " model training function."
+        )
+    clf = linear_model.Lasso(alpha=kwargs["alpha"] if "alpha" in kwargs else 1.0)
+    clf.fit(
+        interpretable_inputs.cpu().numpy(),
+        expected_outputs.cpu().numpy(),
+        weights.cpu().numpy(),
+    )
+    return torch.from_numpy(clf.coef_)
+
+
+def default_from_interp_rep_transform(curr_sample, original_inputs, **kwargs):
+    assert (
+        "feature_mask" in kwargs
+    ), "Must provide feature_mask to use default interpretable representation transform"
+    assert (
+        "baselines" in kwargs
+    ), "Must provide baselines to use default interpretable representation transfrom"
+    feature_mask = kwargs["feature_mask"]
+    if isinstance(feature_mask, Tensor):
+        binary_mask = curr_sample[0][feature_mask]
+        return binary_mask * original_inputs + (1 - binary_mask) * kwargs["baselines"]
+    else:
+        binary_mask = tuple(
+            curr_sample[0][feature_mask[j]] for j in range(len(feature_mask))
+        )
+        return tuple(
+            binary_mask[j] * original_inputs[j]
+            + (1 - binary_mask[j]) * kwargs["baselines"][j]
+            for j in range(len(feature_mask))
+        )
+
+
+def get_similarity_function(
+    distance_mode: str = "cosine", kernel_width: float = 1.0
```

Comment:
I think it might be better to leave the width larger for a default, since if the use-case has lower similarities for most perturbations, this may cause a few perturbations to be highly weighted rather than more equally weighting perturbations, which a new user might not realize and could have misleading results.